### PR TITLE
Full support for highlighting entities.

### LINKED ISSUE #1805: [html] add highlights for entities, &amp; etc.
Highlight js do not highlighted html entities!

Example:
&amp;amp;gt;

could this bug be corrected?

### PR DESCRIPTION
**Pull request by w3suli**
Highlighting entities.
Resolve #1805 issue.

### CODE REVIEW

File: src/languages/xml.js [hunk 1]
```diff
@@ -41,7 +41,11 @@ function(hljs) {
         className: 'meta',
         begin: '<!DOCTYPE', end: '>',
         relevance: 10,
-        contains: [{begin: '\\[', end: '\\]'}]
+        contains: [
+          hljs.QUOTE_STRING_MODE,
+          hljs.APOS_STRING_MODE,
```
Comment (yyyc514) [inline src/languages/xml.js]: What is the thinking with lines 45-46? Is this related to the entity fixes?

**Comment by yyyc514** [comment]
Please provide a markup test to show this new functionality working as expected.  Also a changeling entry would be great.

File: src/languages/xml.js [hunk 2]
```diff
@@ -54,6 +58,10 @@ function(hljs) {
         begin: '<\\!\\[CDATA\\[', end: '\\]\\]>',
         relevance: 10
       },
+      {
+        className: 'symbol',
+        begin: '&[a-z]+;|&#[0-9]+;|&#x[a-f0-9]+;', relevance: 10,
```
Comment (yyyc514) [inline src/languages/xml.js]: This probably should not be relevance 10, many other formats that are HTML variants also have entities... This might be why your relevance tests are breaking.

File: src/languages/xml.js (hunk 1 above)
Comment (w3suli) [reply src/languages/xml.js]: I outsource this section to a separate request.

File: src/languages/xml.js (hunk 2 above)
Comment (w3suli) [reply src/languages/xml.js]: I slightly modified the highlighting of entities. In the new version, entities are already highlighted in the attribute strings.

**Comment by w3suli** [comment]
What is "The Travis CI build failed"?
"should markup template_tags failing"

For me, the Twig temples are flawlessly highlighted.
I did not encounter any errors during the tests!

I need help finding out if the twig test fails. I have tried and have the following code highlighted correctly:
```
{% if posts|length %}
  {% for article in articles %}
  <div>
  {{ article.title|upper() }}

  {# outputs 'WELCOME' #}
  </div>
  {% endfor %}
{% endif %}

{% set user = json_encode(user) %}
```
In the example below, the "div" tag is used as an entity. Then why bother being highlighted as an entity?
```
{% if posts|length %}
  {% for article in articles %}
  &lt;div&gt;
  {{ article.title|upper() }}

  {# outputs 'WELCOME' #}
  &lt;/div&gt;
  {% endfor %}
{% endif %}

{% set user = json_encode(user) %}

{{ random(['apple', 'orange', 'citrus']) }}

{{ include(template_from_string("Hello {{ name }}")) }}


{#
Comments may be long and multiline.
Markup is &lt;em&gt;not&lt;/em&gt; highlighted within comments.
#}
```

entities

Current example (https://highlightjs.org/static/demo/):
entities-orig

**Comment by yyyc514** [comment]
Look at the failing test. 
F81BF260-E304-4EDD-A53B-D809D0BF768C

You need to update the output since it includes entities. Yes?

File: src/languages/xml.js [hunk 3]
```diff
@@ -6,6 +6,10 @@ Category: common
 
 function(hljs) {
   var XML_IDENT_RE = '[A-Za-z0-9\\._:-]+';
+    var XML_ENTITIES = {
+    className: 'symbol',
+    begin: '&[a-z]+;|&#[0-9]+;|&#x[a-f0-9]+;'
+  };
```
Comment (yyyc514) [inline src/languages/xml.js]: Much nicer.

Comment (w3suli) [reply src/languages/xml.js]: Test output corrected.

**Comment by yyyc514** [comment]
Changelog entry?

**Comment by w3suli** [comment]
Changelog updated

**Comment by yyyc514** [comment]
@w3suli many thanks for your first contribution!

**Comment by w3suli** [comment]
@yyyc514 I'm glad I could help.
